src/transformState.js

```diff
@@ -5,7 +5,37 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const extraDatavalue = action.extraData;
+    const typeOfAction = action.type;
+    const removeValue = action.keysToRemove;
+
+    switch (typeOfAction) {
+      case 'addProperties':
+        for (const newProperty in extraDatavalue) {
+          state[newProperty] = extraDatavalue[newProperty];
+        }
+
+        break;
+
+      case 'removeProperties':
+        removeValue.forEach(function(item) {
+          delete state[item];
+        });
+
+        break;
+
+      case 'clear':
+        Object.keys(state).forEach(key => delete state[key]);
+
+        break;
+
+      default:
+        return 'Error!';
```

Comment:
If you want to notify a user about errors it's better to throw exceptions, you can check [docs ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw?retiredLocale=uk) how to use it in js.
Because from this function, we expect returned object "state", but we get a string instead and no errors in this case.